File: clients/algoliasearch-client-php/lib/Model/Recommend/ExactOnSingleWordQuery.php
<?php

namespace Algolia\AlgoliaSearch\Model\Recommend;

use Algolia\AlgoliaSearch\ObjectSerializer;

/**
 * ExactOnSingleWordQuery Class Doc Comment
 *
 * @category Class
 * @description Controls how the exact ranking criterion is computed when the query contains only one word.
 * @package Algolia\AlgoliaSearch
 */
class ExactOnSingleWordQuery
{
    /**
     * Possible values of this enum
     */
    public const ATTRIBUTE = 'attribute';

    public const NONE = 'none';

    public const WORD = 'word';

    /**
     * Gets allowable values of the enum
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [self::ATTRIBUTE, self::NONE, self::WORD];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/main/java/com/algolia/model/search/ExactOnSingleWordQuery.java
package com.algolia.model.search;

import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** Controls how the exact ranking criterion is computed when the query contains only one word. */
public enum ExactOnSingleWordQuery {
  ATTRIBUTE("attribute"),

  NONE("none"),

  WORD("word");

  private final String value;

  ExactOnSingleWordQuery(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static ExactOnSingleWordQuery fromValue(String value) {
    for (ExactOnSingleWordQuery b : ExactOnSingleWordQuery.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}
